test/images/agnhost/bad-tcp-seq-server/tcp.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Partially copied from https://github.com/bowei/lighthouse/blob/master/pkg/probe/tcp.go
+
+package badtcpseqserver
```

Comment:
do we have unit test for this?

Reply:
It would be good to have some sort of test to ensure that the test program as a whole actually disrupts the TCP stream in the way it claims to (see also https://github.com/kubernetes/kubernetes/pull/94135#discussion_r474175888), because if it _fails_ to do that, then the associated e2e test will _pass_ even though it didn't actually test the thing it meant to be testing. If we had that then I don't think unit tests here would be especially necessary.

But this wouldn't be a kube e2e test, it would be a test of agnhost itself, to be run when building a new agnhost release image, which is something I think doesn't exist yet.

(There are unit tests in some of the other subdirs, so we could at least test that this parsing code here works correctly (by testing on previously-tcpdumped data), but the actual server requires `CAP_NET_RAW`, and I don't think it would be possible to test it without `CAP_NET_ADMIN` as well, so I don't think we can do that from unit tests.)